### ISSUE TITLE
stablehlo.convolution with uniform strides/dilations fails: splat dense<1> : tensor<1xi64> lowered instead of tensor<2xi64>

### ISSUE THREAD

**Issue by rivershah** [iree-org/iree#23545]
IREE's StableHLO → linalg lowering produces `strides = dense<1> : tensor<1xi64>` and `dilations = dense<1> : tensor<1xi64>` for 2D
  convolutions with uniform strides/dilations. The `linalg.conv_2d_nhwc_hwcf` verifier rejects this because it requires `tensor<2xi64>`.

  This breaks all standard `keras.layers.Conv2D` and `jax.lax.conv_general_dilated` calls with default strides=(1,1) and no dilation when
  compiled through the JAX → StableHLO → IREE path.

  Conv1D is unaffected because `tensor<1xi64>` happens to be the correct shape for 1 spatial dimension.

  ## Reproducer

  ```mlir
  // conv2d.mlir — exported from JAX 0.8.2 via jax.export
  module @jit_conv2d attributes {jax.uses_shape_polymorphism = false, mhlo.num_partitions = 1 : i32, mhlo.num_replicas = 1 : i32} {
    func.func public @main(%arg0: tensor<3x3x3x16xf32>, %arg1: tensor<2x8x8x3xf32>) -> tensor<2x8x8x16xf32> {
      %0 = stablehlo.convolution(%arg1, %arg0) dim_numbers = [b, 0, 1, f]x[0, 1, i, o]->[b, 0, 1, f], window = {pad = [[1, 1], [1, 1]]}
  {batch_group_count = 1 : i64, feature_group_count = 1 : i64} : (tensor<2x8x8x3xf32>, tensor<3x3x3x16xf32>) -> tensor<2x8x8x16xf32>
      return %0 : tensor<2x8x8x16xf32>
    }
  }
  ```

  ```bash
  iree-compile \
    --iree-hal-target-device=local \
    --iree-hal-local-target-device-backends=llvm-cpu \
    --iree-llvmcpu-target-cpu=host \
    conv2d.mlir -o conv2d.vmfb
  ```

  ## Error

  ```
  error: 'linalg.conv_2d_nhwc_hwcf' op attribute 'strides' failed to satisfy constraint: 64-bit signless int elements attribute of shape [2]
  see current operation:
  %4 = "linalg.conv_2d_nhwc_hwcf"(%3, %arg0, %2) <{
    dilations = dense<1> : tensor<1xi64>,
    strides = dense<1> : tensor<1xi64>
  }> ...
  ```

  ## Expected

  The lowering should produce `dense<1> : tensor<2xi64>` (2-element splat), not `dense<1> : tensor<1xi64>` (1-element tensor).

  ## Observations

  | Case | Result |
  |------|--------|
  | Uniform strides `(1,1)`, uniform dilations `(1,1)` | **Fails** — both are `tensor<1xi64>` |
  | Asymmetric strides `(1,2)`, uniform dilations `(1,1)` | **Fails** — strides OK `tensor<2xi64>`, dilations still `tensor<1xi64>` |
  | Asymmetric strides `(1,2)`, asymmetric dilations `(1,2)` | **Passes** — both are `tensor<2xi64>` |
  | Conv1D with uniform stride `(1,)` | **Passes** — `tensor<1xi64>` is correct for 1 spatial dim |

  ## Versions

  - `iree-base-compiler`: 3.10.0 (`ae97779f59a81bc56f804927be57749bb22548fa`)
  - `jax`: 0.8.2
  - `jaxlib`: 0.8.2
  - LLVM: 22.0.0git
  - Platform: Linux x86_64

**Comment by jschuhmacher** [comment]
Hi @rivershah, thanks for this detailed reproducer! I'm seeing the same behavior, I can take a look.

**Comment by jschuhmacher** [comment]
The root cause for the failure lies in the StableHLO to linalg conversion. StableHLO had a small bug in handling its default values.

**Comment by jschuhmacher** [comment]
The fix is merged in openxla/stablehlo. Next step is to propagate it here.

**Comment by hanhanW** [comment]
What is the status of this issue?

**Comment by jschuhmacher** [comment]
It's fixed and merged in upstream StableHLO. It looks like the last integration is just 2 commits short of having the fix. I'll open a PR to integrate it.
